### PR TITLE
Minor improvements to PrioritizedThrottledTaskRunnerTests

### PR DESCRIPTION
**Pull request by DaveCTurner**
- Makes the conditions on which we're waiting a little more precise, so there's no need to `assertBusy()`.

- Makes the tests a little lighter by not using the large `GENERIC` threadpool executor.

- Adjusts `testTasksRunInOrder` to wait for the enqueueing to finish using a task rather than overriding the (now-private) `pollAndSpawn()` method.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
Consider this whole PR as just suggestions/discussion points - none of this is a huge deal here, just stuff I noticed while I was passing through.

File: server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java [hunk 1]
```diff
@@ -69,119 +73,120 @@ public void onFailure(Exception e) {
...
+                    executedCountDown.countDown();
+                }, priority));
+                awaitBarrier(enqueuedBarrier);
+            }).start();
         }
-        assertBusy(() -> assertThat(taskRunner.queueSize(), equalTo(enqueued)));
-        assertThat(taskRunner.runningTasks(), equalTo(0));
-        workerBlocker.countDown();
+        awaitBarrier(enqueuedBarrier);
+        awaitBarrier(enqueuedBarrier);
```
Comment (pxsalehi) [inline server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java:137]: The distinction between the two awaits here are not very clear to understand. One is to unblock the enqueued tasks at L108? Could you add a comment here please.

**Review by pxsalehi — APPROVED** [review]
LGTM (assuming green CI). Thanks for the PR David!

File: server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java (hunk 1 above)
Comment (pxsalehi) [reply server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java:137]: Maybe a more concrete question would be: why are we sync'ing all the threads? Wouldn't all the tasks enqueued in the threads be blocked anyway?

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java]: Really this is just so the `enqueueTask()` calls are as concurrent as possible. Without this, we're relying on going round the thread-creation loop quickly enough to generate concurrent activity. I expect we'd catch issues eventually either way, but empirically this seems to make failures a little easier to reproduce.

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java]: I added clarifying comments in 00ef643f835, hope that's ok.

**Pull request merged**